index.ts

```diff
@@ -0,0 +1,32 @@
+import { UserSerice } from "./src/users";
+import { TaskService } from "./src/tasks";
+
+const task = new TaskService();
+const user = new UserSerice(task);
```

Comment:
esos se llamarian `....Service` tambien, ya que son la instancia, no es una task ni un user, son los servicios

```suggestion
const taskService = new TaskService();
const userService = new UserSerice(taskService);
```